File: src/test/java/uk/gov/justice/hmpps/prison/service/transformers/AgencyTransformerTest.java
package uk.gov.justice.hmpps.prison.service.transformers;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import uk.gov.justice.hmpps.prison.api.model.Agency;
import uk.gov.justice.hmpps.prison.repository.jpa.model.ActiveFlag;
import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyLocation;
import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyLocationType;

import static org.assertj.core.api.Assertions.assertThat;
import static uk.gov.justice.hmpps.prison.service.transformers.AgencyTransformer.transform;

public class AgencyTransformerTest {

    private final AgencyLocation.AgencyLocationBuilder builder = AgencyLocation.builder();

    @BeforeEach
    void setup() {
        builder.id("MDI").type(new AgencyLocationType("INST")).description("moorland");
    }

    @Test
    void transform_active_and_description_capitalisation() {
        var agency = builder.activeFlag(ActiveFlag.Y).build();

        assertThat(transform(agency)).isEqualTo(
                Agency.builder()
                        .agencyId("MDI")
                        .agencyType("INST")
                        .description("Moorland")
                        .active(true)
                        .build());
    }

    @Test
    void transform_inactive() {
        var agency = builder.activeFlag(ActiveFlag.N).build();

        assertThat(transform(agency)).isEqualTo(
                Agency.builder()
                        .agencyId("MDI")
                        .agencyType("INST")
                        .description("Moorland")
                        .active(false)
                        .build());
    }

    @Test
    void transform_active_unspecified() {
        var agency = builder.build();

        assertThat(transform(agency)).isEqualTo(
                Agency.builder()
                        .agencyId("MDI")
                        .agencyType("INST")
                        .description("Moorland")
                        .active(false)
                        .build());
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/service/transformers/AgencyTransformerTest.kt
package uk.gov.justice.hmpps.prison.service.transformers

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import uk.gov.justice.hmpps.prison.api.model.Agency
import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyLocation
import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyLocationType

class AgencyTransformerTest {
  private val builder: AgencyLocation.AgencyLocationBuilder = AgencyLocation.builder()

  @BeforeEach
  fun setup() {
    builder.id("MDI").type(AgencyLocationType("INST")).description("moorland")
  }

  @Test
  fun transform_active_and_description_capitalisation() {
    val agency = builder.active(true).build()

    assertThat(AgencyTransformer.transform(agency, false, false)).isEqualTo(
      Agency.builder()
        .agencyId("MDI")
        .agencyType("INST")
        .description("Moorland")
        .active(true)
        .build(),
    )
  }

  @Test
  fun transform_inactive() {
    val agency = builder.active(false).build()

    assertThat(AgencyTransformer.transform(agency, false, false)).isEqualTo(
      Agency.builder()
        .agencyId("MDI")
        .agencyType("INST")
        .description("Moorland")
        .active(false)
        .build(),
    )
  }

  @Test
  fun transform_active_unspecified() {
    val agency = builder.build()

    assertThat(AgencyTransformer.transform(agency, false, false)).isEqualTo(
      Agency.builder()
        .agencyId("MDI")
        .agencyType("INST")
        .description("Moorland")
        .active(false)
        .build(),
    )
  }
}
